ginga/Control.py

```diff
@@ -517,29 +514,37 @@ def load_image(self, filepath, idx=None):
             filepath = filepfx
         else:
             inherit_prihdr = self.settings.get('inherit_primary_header', False)
-            image = AstroImage.AstroImage(logger=self.logger,
-                                          inherit_primary_header=inherit_prihdr)
+            image = AstroImage.AstroImage(
+                logger=self.logger, inherit_primary_header=inherit_prihdr)
             kwdargs.update(dict(numhdu=idx))
 
         try:
             self.logger.info("Loading image from %s kwdargs=%s" % (
                 filepath, str(kwdargs)))
             image.load_file(filepath, **kwdargs)
 
-        except Exception as e:
-            errmsg = "Failed to load file '%s': %s" % (
-                filepath, str(e))
-            self.logger.error(errmsg)
+        except Exception:
+            # TODO: Refactor code to be less hacky.
+            # Maybe it has a table on EXT 1.
             try:
```

Comment:
My suggestion would be: once we have determined the file is a FITS file (or we have determined that we don't know what it is, so we are assuming it is FITS), we ask `ginga.util.io_fits` to open it and magically return us the right kind of thing (`AstroImage` or `AstroTable`).  We can assume that module has more detailed info than we have here; for example, once it opens the file and takes a look at the HDU it will know what kind of thing to create.